Entities/AHuman.cpp

```diff
@@ -3125,66 +3125,70 @@ void AHuman::Update()
 	bool isStill = (m_Vel + m_PrevVel).MagnitudeIsLessThan(movementThreshold);
 	bool isSharpAiming = m_Controller.IsState(AIM_SHARP);
 
-	// If the pie menu is on, try to preserve whatever move state we had before it going into effect.
-	if (!m_Controller.IsState(PIE_MENU_ACTIVE)) {
-		bool crouching = m_Controller.IsState(BODY_CROUCH);
-		if ((m_Controller.IsState(MOVE_RIGHT) || m_Controller.IsState(MOVE_LEFT) || m_MoveState == JUMP) && m_Status != INACTIVE) {
-			for (int i = WALK; i < MOVEMENTSTATECOUNT; ++i) {
-				m_Paths[FGROUND][i].SetHFlip(m_HFlipped);
-				m_Paths[BGROUND][i].SetHFlip(m_HFlipped);
-			}
-			// Only if not jumping, OR if jumping, and apparently stuck on something - then help out with the limbs.
-			if (m_MoveState != JUMP || isStill) {
-				// Restart the stride if we're just starting to walk or crawl.
-				if ((m_MoveState != WALK && !crouching) || (m_MoveState != CRAWL && crouching)) {
-					m_StrideStart = true;
-					MoveOutOfTerrain(g_MaterialGrass);
-				}
+    // If the pie menu is on, try to preserve whatever move state we had before it going into effect.
+    // This is only done for digital input, where the user needs to use the keyboard to choose pie slices
+    // For analog input, this doesn't matter - the mouse or aiming analog stick controls the pie menu
+    bool keepOldState = !m_Controller.IsMouseControlled() && !m_Controller.IsGamepadControlled() && m_Controller.IsState(PIE_MENU_ACTIVE);
```

Comment:
I dunno if you really need to pull this one out into a variable, especially with the thiccc set of comments above it haha.
Oh, also, comments should end with a `.` btw

Reply:
I prefer to be explicit I guess. I don't think it hurts to have these comments and the variable both - I want to make it clear that the semantic action (keeping old movement state) is different from the reason _why_

Reply:
I think the comments you added are worthwhile. The original dan one which explains what the code is doing, less so haha, it could/should be replaced by a private method that explains things just as clearly (but don't bother really, AHuman needs a ground up rewrite someday).

I don't really see the benefit of the separate variable for this, especially since your comments explain the why of things, and the code can explain the what. But, I also won't make a fuss, leave it be if you prefer it that way :)